x-pack/plugins/endpoint/server/routes/index_pattern.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { IRouter, Logger, RequestHandler } from 'kibana/server';
+import { EndpointAppContext } from '../types';
+import { IndexPatternGetParamsResult, EndpointAppConstants } from '../../common/types';
+import { indexPatternGetParamsSchema } from '../../common/schema/index_pattern';
+import { IndexPatternRetriever } from '../index_pattern';
+
+function handleIndexPattern(
+  log: Logger,
+  indexRetriever: IndexPatternRetriever
+): RequestHandler<IndexPatternGetParamsResult> {
+  return async (context, req, res) => {
+    try {
+      return res.ok({
+        body: {
+          indexPattern: await indexRetriever.getIndexPattern(context, req.params.datasetPath),
+        },
+      });
+    } catch (error) {
+      log.warn(error);
+      return res.notFound({ body: error });
+    }
+  };
+}
+
+export function registerIndexPatternRoute(router: IRouter, endpointAppContext: EndpointAppContext) {
```

Comment:
is this for the alert list feature? if so, thoughts on moving it to the alerting routes module?

Reply:
Yeah it's currently only used for the alert list. The alert list uses an index pattern when doing time based filter I think.

I could foresee other pages (Host maybe?) needing an index pattern if they want to do any time based filtering too. If you'd prefer it in the alert list for now until it's needed in the future, that's fine too though.